### ISSUE TITLE
v0.4.0 small bug: "name" is ignored when upgrading existing vendor

### ISSUE THREAD

**Issue by bwplotka** [jsonnet-bundler/jsonnet-bundler#116]
Repro:

*  Have jsonnetfile.json with

```
{
      "source": {
        "git": {
          "remote": "https://github.com/thanos-io/thanos.git",
          "subdir": "mixin/thanos"
        }
      },
      "version": "master",
      "name": "thanos-mixin"
    }
```

* Run `jb install`

We should see (broken!) `thanos-mixin -> github.com/thanos-io/thanos/mixin/thanos` link. (BTW can we have error message when that happens? jb install should never succeed then.

* Edit `jsonnetfile.json` to 

```
{
      "source": {
        "git": {
          "remote": "https://github.com/thanos-io/thanos.git",
          "subdir": "mixin"
        }
      },
      "version": "master",
      "name": "thanos-mixin"
    }
```
* Run `jb install`

in vendor we can see link `mixin -> github.com/thanos-io/thanos/mixin` , despite the value in `name`... We expect: `thanos-mixin -> github.com/thanos-io/thanos/mixin`

---

If I do `rm -rf vendor` and `jb install` all works I can see  `thanos-mixin -> github.com/thanos-io/thanos/mixin`

**Comment by metalmatze** [comment]
Makes sense. I'm not sure if I want to spend the time trying to fix a legacy thing (symlink). 
If it's broken for you, then I'd say, go ahead and start using the absolute import paths already. :innocent:
